src/main.rs

```diff
@@ -70,36 +84,60 @@ fn main() {
 }
 
 fn run() -> Result<(), MainError> {
-    let args: Vec<String> = std::env::args().collect();
-    match &*args[1] {
+    let cli = Cli::parse();
+    match &*cli.name[0] {
         "run" | "run-script" => {
-            let manifest = "package.json"
+            let mut cwd = env::current_dir().expect("Couldn't find the current working directory");
+            if cli.workspace_root {
+                cwd = find_workspace_root(&cwd)?;
+            }
+            let manifest_path = cwd.join("package.json");
+            let manifest = manifest_path
                 .pipe(File::open)
                 .map_err(MainError::from_dyn)?
                 .pipe(serde_json::de::from_reader::<_, NodeManifest>)
                 .map_err(MainError::from_dyn)?;
-            let name = args[2].clone();
+            let name = cli.name[1].clone();
             if let Some(command) = manifest.scripts.get(&name) {
                 eprintln!("> {:?}", command);
-                run_script(name, command.clone())
+                run_script(name, command.clone(), &cwd)
             } else {
                 PnError::MissingScript { name }
                     .pipe(MainError::Pn)
                     .pipe(Err)
             }
         }
-        "install" | "i" | "update" | "up" => pass_to_pnpm(&args[1..]),
-        _ => pass_to_sub(args[1..].join(" ")),
+        "install" | "i" | "update" | "up" => pass_to_pnpm(&cli.name[0..]),
+        _ => pass_to_sub(cli.name[0..].join(" ")),
+    }
+}
+
+fn find_workspace_root(cwd: &dyn AsRef<Path>) -> Result<PathBuf, MainError> {
+    let workspace_manifest_location = find_up_with(
+        "pnpm-workspace.yaml",
+        FindUpOptions {
+            kind: FindUpKind::File,
+            cwd: cwd.as_ref(),
+        },
+    )
+    .map_err(MainError::from_dyn)?;
+    match workspace_manifest_location {
+        Some(path) => Ok(path
+            .parent()
+            .ok_or(MainError::Pn(PnError::NotInWorkspace))?
```

Comment:
I am not sure about this line. I don't think this is the right error but I also don't think this error will ever happen.